### ISSUE TITLE
[🐛 Bug]: Firefox hangs when testing after upgrade to Selenium 4.x Python binding

### ISSUE THREAD

**Issue by dornech** [SeleniumHQ/selenium#10557]
### What happened?

during tests after upgrading to Selenium 4.x I encounter problems with FireFox. the Firefox browser hangs after being started and the Python webdriver object does not return from initialization. Same code works fine with Chrome. Differences in driver implementation i e addon installation are not relevant in the test i e I do not use any specific setting, user profile or something else. 
Latest Geckodriver and Firefox.
Geckodriver process as well as a Firefox (with this robot icon in the address bar proving it being started remotely) are started, so the path issue reported often is not an issue (while path is set anyway).
I did a complete reinstall of Firefox *without* taking over old settings, so old firefox settings cannot be the root cause. 
I discovered the issue when recently doing the upgrade to Selenium 4.x and testing some of my standard modules. However, I do not use FireFox very often so I cannot completely rule out it would have happened with 3.141 since a specific firefox version.

Does anybody have similar problems?

### How can we reproduce the issue?

```shell
Standard start, no really extraordinary coding
```

### Relevant log output

```shell
no log output
```

### Operating System

Windows 10

### Selenium version

Python 3.9.5, Selenium 4.1.3

### What are the browser(s) and version(s) where you see this issue?

latest Chrome and Firefox

### What are the browser driver(s) and version(s) where you see this issue?

Geckodriver 0.30 and 0.31

### Are you using Selenium Grid?

No

**Comment by symonk** [comment]
Hi @dornech can you please share a minimum reproducible script with me to investigate?

**Comment by titusfortner** [comment]
Lots of people are using Python with Selenium 4, so we need more information to help you with your issue.

You can set logs for Firefox, which will output to `geckodriver.log`

```python
    options = FirefoxOptions()
    logging.basicConfig(level=logging.DEBUG)
    options.log.level = "TRACE"
    driver = Firefox(options= options)
```

**Comment by dornech** [comment]
Argh, found the problem.

I use a own developed utility package with a webdriver initialization to
allow standard initializations like setting download path, install add
ons and hiding browser specific implementation details from the calling
application.
When extracting the code I found one statement I oversaw so far:

browsersettings.set_preference("xpinstall.signatures.required", False)

I think it worked in Selenium 3.x or at least did not cause trouble when
I worked with it the last time; but I remember I had issues with addons
with Firefox earlier.
Whatsoever - commenting this line I do not have the problem any more.

Thanks for your support and sorry for any confusion caused !



Am 19.04.2022 um 19:30 schrieb Titus Fortner:
